pkg/services/alerting/extractor.go

```diff
@@ -170,7 +174,13 @@ func (e *DashAlertExtractor) getAlertFromPanels(ctx context.Context, jsonWithPan
 			panelQuery := findPanelQueryByRefID(panel, queryRefID)
 
 			if panelQuery == nil {
-				reason := fmt.Sprintf("Alert on PanelId: %v refers to query(%s) that cannot be found", alert.PanelId, queryRefID)
+				config := ctx.Value(CtxCfg{}).(*setting.Cfg)
+				var reason string
+				if config.UnifiedAlerting.Enabled {
+					reason = fmt.Sprintf("Alert on PanelId: %v refers to query(%s) that cannot be found. Legacy alerting queries are not able to be removed at this time in order to preserve the ability to rollback to previous versions of Grafana", alert.PanelId, queryRefID)
```

Comment:
```suggestion
					reason = fmt.Sprintf("alertId '%v' on PanelId: '%v' refers to query(%s) that cannot be found. Legacy alerting conditions are not able to be removed at this time in order to preserve the ability to rollback to previous versions of Grafana", alert.Id, alert.PanelId, queryRefID)
```

Reply:
Here, the error is specifically on removing the query so I think we should not reference condition

Reply:
🤦 - you're right.